src/osx/textctrl_osx.cpp

```diff
@@ -229,11 +229,19 @@ wxSize wxTextCtrl::DoGetBestSize() const
          hText *= 5 ;
 
     if ( !HasFlag(wxNO_BORDER) )
-        hText += 6 ;
+        hText += 5 ;
 
     return wxSize(wText, hText);
 }
 
+wxSize wxTextCtrl::DoGetSizeFromTextSize(int xlen, int ylen) const
+{
+    wxSize size = wxDefaultSize;
+    if (xlen > 0) size.SetWidth(HasFlag(wxNO_BORDER) ? xlen + 4 : xlen + 9);
+    if (ylen > 0) size.SetHeight(HasFlag(wxNO_BORDER) ? ylen + 2 : ylen + 7);
```

Comment:
I understand that the difference between 2 values is `BORDER_SIZE` constant (5), but does anyone know/understand/can explain where do the border-less values come from, i.e. why do we need 4 extra px horizontally and 2 px vertically?

Reply:
we had the problem that the focus rect is painted outside the textcontrol and we ran into redraw problems in certain situations after focus rects

Reply:
The way I measure these values is:
* Choose a certain string, like "xxxx" or "WWWW"
* Use `GetTextExtent` to get the text size
* Increase the TextCtrl's size until the text won't scroll when moving cursor or being selected

![wx](https://user-images.githubusercontent.com/8182182/82355243-f3748080-99cf-11ea-9193-30f137b45125.gif)'
Above is an image of `wxTextCtrl(wxNO_BORDER)` and a normal `wxTextCtrl`. The `wxNO_BORDER` one still has a thin border (1px on each side), and a 1px margin on left and right (can be seen when the text is selected).

Reply:
I see, thanks. What worries me is that it looks like these values could be different on other macOS versions, so I hoped that perhaps there would be some way to get these values via the API.

But if not, we'll just have to live with this, of course. Could you please just add a comment to the code (or even just reply here) saying under which macOS version have you measured these values? TIA!

Reply:
> What worries me is that it looks like these values could be different on other macOS versions

I guess the previous values are just for an older version of macOS. I'm new to Cocoa or mac GUI, so I'm not sure if there is such an API. I'm looking into `NSLayoutManager`, and any suggestions are appreciated.

Reply:
Values in this PR are measured on macOS Catalina 10.15.5 19F83c.